pkg/config/config.go

```diff
@@ -92,53 +134,48 @@ const (
 	CaptureStderrFiles
 )
 
-// Validate does static validation of the configuration
-func (tc Config) Validate() error {
-	for p := range tc.Policies.Map() {
-		if p == nil {
-			return errfmt.Errorf("policy is nil")
-		}
-		if p.EventsToTrace == nil {
-			return errfmt.Errorf("policy [%d] has no events to trace", p.ID)
-		}
+//
+// Capabilities
+//
 
-		for e := range p.EventsToTrace {
-			_, exists := events.Definitions.GetSafe(e)
-			if !exists {
-				return errfmt.Errorf("invalid event [%d] to trace in policy [%d]", e, p.ID)
-			}
-		}
-	}
-	if (tc.PerfBufferSize & (tc.PerfBufferSize - 1)) != 0 {
-		return errfmt.Errorf("invalid perf buffer size - must be a power of 2")
-	}
-	if (tc.BlobPerfBufferSize & (tc.BlobPerfBufferSize - 1)) != 0 {
-		return errfmt.Errorf("invalid perf buffer size - must be a power of 2")
-	}
-	if len(tc.Capture.FileWrite.PathFilter) > 3 {
-		return errfmt.Errorf("too many file-write path filters given")
-	}
-	for _, filter := range tc.Capture.FileWrite.PathFilter {
-		if len(filter) > 50 {
-			return errfmt.Errorf("the length of a path filter is limited to 50 characters: %s", filter)
-		}
-	}
-	if len(tc.Capture.FileRead.PathFilter) > 3 {
-		return errfmt.Errorf("too many file-read path filters given")
-	}
-	for _, filter := range tc.Capture.FileWrite.PathFilter {
-		if len(filter) > 50 {
-			return errfmt.Errorf("the length of a path filter is limited to 50 characters: %s", filter)
-		}
-	}
+type CapabilitiesConfig struct {
+	BypassCaps bool
+	AddCaps    []string
+	DropCaps   []string
+}
 
-	if tc.BPFObjBytes == nil {
-		return errfmt.Errorf("nil bpf object in memory")
-	}
+//
+// Output
+//
 
-	if tc.ChanEvents == nil {
-		return errfmt.Errorf("nil events channel")
-	}
+type OutputConfig struct {
+	StackAddresses bool
+	ExecEnv        bool
+	RelativeTime   bool
+	ExecHash       bool
 
-	return nil
+	ParseArguments    bool
+	ParseArgumentsFDs bool
+	EventsSorting     bool
+}
+
+type ContainerMode int
+
+const (
+	ContainerModeDisabled ContainerMode = iota
+	ContainerModeEnabled
+	ContainerModeEnriched
+)
+
+type PrinterConfig struct {
+	Kind          string
+	OutPath       string
+	OutFile       io.WriteCloser
+	ContainerMode ContainerMode
+	RelativeTS    bool
+}
+
+type PrinterOutputConfig struct {
```

Comment:
This is the only one I'm not sure. Because it is not actually a config, it was created just to simplify the return from `PrepareOutput` to return two values only, instead of three which makes the code look very ugly. Maybe it should be a structure there, for it only still? `type PrepareOutputResult struct{}`? WDYT?

```suggestion
type PrinterOutputConfig struct {
```

Reply:
> This is the only one I'm not sure. Because it is not actually a config, it was created just to simplify the return from `PrepareOutput` to return two values only, instead of three which makes the code look very ugly.

Oh I understand. In this case I'll leave it as it is.

> Maybe it should be a structure there, for it only still? `type PrepareOutputResult struct{}`? WDYT?

Can it be internal to the pkg? prepareOutputResult?

Reply:
@geyslan yup, it can be a internal one

Reply:
Actually it can't since it's used from cobra and urfave, so I've just renamed as you suggested. 👍🏼